### PR TITLE
Add Minishell write up from THCon 2024

### PR DESCRIPTION
**Pull request by ctmbl**

### CODE REVIEW

**Review by ZynoXelek — COMMENTED** [review]
Once these two points will be answered, the article is really clean and ready to be uploaded!

Just another quick question : As I did not look into the code of this challenge, could a link to the code be added for people that did not personnaly look into it?

File: src/content/posts/minishell_wu_pwn.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+`log.c` is really simple, just a `main` function, it's a logging tool, it will write its arguments passed in command line to a log file, that's all.
+
+`minishell.c` is really something else: 269 lines of code.  
+When reading `C` code I always start looking globally at the function names and then I deep into the `main` function first.
+Here it helped a lot, in `main` we quickly note that there is a bunch of variables initialization, some memory allocation and then a `while(1)`!
+This is the infinite loop allowing the shell to always wait for user instructions.
+
+We understand that the user is prompted for a string, which is then verified (some characters are forbidden in `commandAllowed` maybe there is something here) and parsed with `strtok`.
+Then a bunch of `if else` identify which function to execute given the user command. At that point I could have start looking into each and every function to look for vulnerabilities.
+But I didn't, I wanted to finish first the reading of `main` and I chose really well.
```
[the remark this answers is not in the document]
Comment (ctmbl) [reply src/content/posts/minishell_wu_pwn.md:39]: in fact this is better tks!

File: src/content/posts/minishell_wu_pwn.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+> **Note**:  
+> However I noticed first that at the beginning of the `while` loop `buffer` is copied into `cmd` **before** verifying it with `commandAllowed`.  
+> So I tested an exploit where I injected some forbidden command `aa; /bin/sh` which won't be executed **but will be written in `cmd` anyway**.  
+> And then I inject a second one `a` which is allowed but unrecognized: the idea was that it didn't totally overwrite `cmd` which then would be something like `a\n; /bin/sh` and be appended to `log` then executed.  
+> Unfortunately, `strcpy` (or other reason) adds a `\x00` between the "new" injection `a` and the "remaining" one in `cmd`, so it ends the string and even if the payload is there in the stack it isn't copied in `log` and wouldn't have been executed by `system` anyway.  
+> So close!
+
+So the real vulnerability is still here lying under our eyes: simply `arg` is not the same size as `cmd`, then when copying a long `cmd` into `arg` it overflows.
```
[the remark this answers is not in the document]
Comment (ctmbl) [reply src/content/posts/minishell_wu_pwn.md:67]: nop this note was an idea which turn out to be a dead end so this is normal 😉

[the remark this answers is not in the document]
Comment (ZynoXelek) [reply src/content/posts/minishell_wu_pwn.md:67]: Then, line 73 and 74 should be inside the note ?

[the remark this answers is not in the document]
Comment (ctmbl) [reply src/content/posts/minishell_wu_pwn.md:67]: @ZynoXelek

[the remark this answers is not in the document]
Comment (ZynoXelek) [reply src/content/posts/minishell_wu_pwn.md:67]: Ok, then everything is good!

**Comment by ctmbl** [comment]
@ZynoXelek 

Sorry didn't see your request, actually there is a link at the beginning, to my github containing the source code but I could add a more explicit one just before diving in the code maybe?

**Review by ZynoXelek — APPROVED** [review]
Very good post in my opinion!

**Comment by ctmbl** [comment]
tks @ZynoXelek !

**Pull request merged**